src/main/java/io/debezium/connector/vitess/connection/VStreamOutputMessageDecoder.java

```diff
@@ -322,10 +328,11 @@ private void handleFieldMessage(Binlogdata.VEvent vEvent) {
                         LOGGER.error("Cannot resolve JDBC type from vstream type {}", columnType);
                     }
                     // TODO: Vitess community has recently added pk flag
-                    boolean key = false;
+                    boolean key = (field.getFlags() & PRI_KEY_FLAG) != 0;
```

Comment:
I'm wondering how that matches with what you say about flag not exposing enough info for multi-column PKs, given that this flag is obtained from the field?

Reply:
Actually, multi-column PKs are supported, single-column unique keys are also supported. Only multi-column unique keys are not supported. The following are some examples of the column [flag](https://dev.mysql.com/doc/dev/mysql-server/8.0.12/group__group__cs__column__definition__flags.html) provided by Vitess:

If a table has a single-column PK, the pk column's flag value in binary format is: `0010`.

If a table has multi-column PK, all columns of the PK has the same flag value in binary format: `0010`.

All single-column unique keys have the same flag value in binary format: `0100`.

The first column in a multi-column unique key has the flag value in binary format: `1000`. The second (and 3rd, etc) column in a multi-column unique key has the flag value in binary format: `0000`. For example:
```
DROP TABLE IF EXISTS t;
CREATE TABLE t
(
    id             BIGINT NOT NULL,
    int_col        INT,
    int_col2       INT,
    int_col3       INT,
    int_col4       INT,
    UNIQUE KEY unique_col_1_2 (int_col, int_col2),
    UNIQUE KEY unique_col_3_4 (int_col3, int_col4)
);
```
`int_col` has flag value `1000`, `int_col2` has flag value `0000`, `int_col3` has flag value `1000`, `int_col4` has flag value `0000`. So there's no easy way to create the Kafka message key from multi-column unique keys.

Reply:
I've just refactored this part by using an enum, hopefully, it is a bit more clear now.